src/tests/ftest/util/code_coverage_utils.py

```diff
@@ -0,0 +1,117 @@
+"""
+  (C) Copyright 2022-2023 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+from logging import getLogger
+import os
+
+from collection_utils import archive_files
+from run_utils import run_remote
+
+
+class CodeCoverage():
+    """Test bullseye code coverage class."""
+
+    def __init__(self, test_env):
+        """Initialize a CodeCoverage object.
+
+        Args:
+            test_env (TestEnvironment): the current test environment.
+        """
+        self.__hosts = None
+        self.__test_env = test_env
+
+    def check(self, hosts):
+        """Determine if bullseye code coverage collection is enabled.
+
+        Args:
+            hosts (NodeSet): hosts on which to check for bullseye code coverage source files
+        """
+        log = getLogger()
+        log.debug("Checking for bullseye code coverage configuration")
+        result = run_remote(hosts, " ".join(["ls", "-al", self.__test_env.bullseye_src]))
+        if not result.passed:
+            log.info("Bullseye code coverage collection not configured on %s", result.failed_hosts)
+            self.__hosts = None
+        else:
+            log.info("Bullseye code coverage collection configured on %s", hosts)
+            self.__hosts = hosts
+
+    def setup(self, result):
+        """Set up the hosts for bullseye code coverage collection.
+
+        Args:
+            result (TestResult): the test result used to update the status of the runner
+
+        Returns:
+            bool: False if there is a problem setting up bullseye code coverage; True otherwise
+
+        """
+        if self.__hosts:
```

Comment:
I think this is a cleaner practice
```suggestion
        if not self.__hosts:
            return True
```
Then the rest of the function doesn't need to be indented